src/classes/CON_Email.cls

```diff
@@ -176,26 +178,29 @@ public class CON_Email {
 
             this.allFields = new List<CON_EmailField>();
             this.valuedFields = new List<CON_EmailField>();
-            Map<String, Schema.DescribeFieldResult> fieldMap = UTIL_Describe.getFieldsOfType('Contact', 'EMAIL');
-            this.emailLabel = fieldMap.get('Email').getLabel();
+
+            if (contactEmailFieldMap == null) {
+                contactEmailFieldMap = UTIL_Describe.getFieldsOfType('Contact', 'EMAIL');
+            }
+            this.emailLabel = contactEmailFieldMap.get('Email').getLabel();
 
             Map<String, String> hedaEmailField = new Map<String, String>();
             //build map for HEDA email field map (Label, API Name)
-            for(String fieldName : fieldMap.keySet() ) {
+            for(String fieldName : contactEmailFieldMap.keySet() ) {
                 if(String.isNotBlank(UTIL_Namespace.getNamespace()) && fieldName.startsWithIgnoreCase(UTIL_Namespace.getNamespace())) {
-                    hedaEmailField.put(fieldMap.get(fieldName).getLabel(), fieldName);
+                    hedaEmailField.put(contactEmailFieldMap.get(fieldName).getLabel(), fieldName);
                 }
             }
 
-            for(String fieldName : fieldMap.keySet() ) {
+            for(String fieldName : contactEmailFieldMap.keySet() ) {
                 //If the field is not a standard email field and it is not duplicate with HEDA email, then add it to the list
-                if ( fieldName != emailLabel && !(hedaEmailField.get(fieldMap.get(fieldName).getLabel()) != null &&
-                                fieldName != hedaEmailField.get(fieldMap.get(fieldName).getLabel()))) {
+                if ( fieldName != emailLabel && !(hedaEmailField.get(contactEmailFieldMap.get(fieldName).getLabel()) != null &&
+                                fieldName != hedaEmailField.get(contactEmailFieldMap.get(fieldName).getLabel()))) {
                     String emailField = (String)contact.get(fieldName);
 
-                    allFields.add( new CON_EmailField( emailField, fieldMap.get(fieldName).getLabel(), fieldName ) );
+                    allFields.add( new CON_EmailField( emailField, contactEmailFieldMap.get(fieldName).getLabel(), fieldName ) );
```

Comment:
if you do what I mentioned above by putting things like contactEmailFieldMap.get(fieldName).getLabel() into a variable then you can re-use it throughout this method. I feel it makes it more readable. if you're uber concerned about performance in so doing however, let me know and we can chat.

Reply:
Definitely not concerned about performance - I think storing it in a variable is more efficient.